presto-main/src/main/java/com/facebook/presto/execution/SqlQueryManager.java

```diff
@@ -335,6 +335,7 @@ public QueryId createQueryId()
 
             // decode session
             session = sessionSupplier.createSession(queryId, sessionContext);
+            accessControl.checkQueryIntegrity(session.getIdentity(), query);
```

Comment:
Do you know what is the approx. latency penalty we will need to pay for this?

Reply:
It should be no-op from WPS if there is no CAT token provided (so similar to any permission check we have today). It usually should be below millisecond to 10s of millisecond in thrift call round trip if I remember correctly. There was some discussion caching the checks to avoid frequent calls to WPS but so far that hasn't seem to be a huge problem (There are some presto use cases makes such calls thousands times per seconds and have <100ms latency). 

But I am not quite sure how much penalty we will pay for those enabled (it will depends on the implementation) or whether we even need to send full query but some digest.